api-client/src/main/java/com/xing/api/XingApi.java

```diff
@@ -115,9 +115,14 @@
         this.resourceFactories = new LinkedHashSet<>(resourceFactories);
         this.resourceFactories.addAll(BUILT_IN_FACTORIES);
     }
-
-    /** Return a {@link Resource} instance specified by the provided class. */
-    @SuppressWarnings("unchecked")
+    
+    /**
+     * Return a {@link Resource} instance specified by the provided class.
+     *
+     * @deprecated This function might use reflection if the factory is not provided, 
+     * better pass this object to the resource to be created instead.
+     */
+    @Deprecated @SuppressWarnings("unchecked")
```

Comment:
Is there an alternative solution already then?

Reply:
If I'm not mistaken this is used in order to cache Resources inside this class, right? If so, an alternative would be to simply use dagger to cache what needs to be cached instead of having this implicit behaviour here.

Maybe you remember better if this function had any other benefit apart from that :)

Reply:
Oh, yeah, I see your point now. 

Although, I believe that is an important change on how library works. Maybe a little bit more explanation about it would be better. Because I believe the original idea was to handle caching as part of library. 

- If the plan to remove this functionality (which means removing the resource caching all together) maybe mention that.
- If it is only about to deprecate, and then maybe explain how would be the better approach - for documentation purposes. I assume by this change, you mean instead of getting the resource instance as `xingApi.resource(WhateverResource.class)` should be created as `WhateverResource(xingApi)` and caching should be handled by developer themselves.

Reply:
Also you mentioned as "it might use reflection if factory is not provided" - while I understand that can be a hidden cost, it is also a design decision, isn't it? You might wanna change that part instead of deprecating the caching mechanism then 🤷‍♂ 

How about throwing an `IllegalStateException`?

Reply:
Sorry for the delay, we've opted for removing this completely, it will be a breaking change but we consider it important in order to prevent issues with Proguard in the future.